.github/workflows/test.yml

```diff
@@ -0,0 +1,47 @@
+name: Run tests
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+jobs:
+    test:
+      runs-on: ubuntu-latest
```

Comment:
Just curious can we use fedora here or do we have to use ubuntu?

Reply:
github runners are all ubuntu afaik, so unless we want to host our own, I believe this is the only option.  If we find there is a different way later, I am not opposed.

Reply:
Just to point to something specific, here are the supported github runners:
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources